crates/revm/src/processor.rs

```diff
@@ -267,7 +267,7 @@ where
             fill_op_tx_env(self.evm.tx_mut(), transaction, sender, envelope_buf.into());
         }
 
-        let hash = transaction.hash();
+        let hash = transaction.recalculate_hash();
```

Comment:
let's recalculate the hash on-demand, i.e. only on the line 278 (in `tracing::trace` call) and on the line 288 (when the error has occurred). Otherwise, it may be not even used, but the recalculation is expensive.

Also, we might want to check if the hash is empty now, because AFAIK it's only an issue during the pipeline syncv.